server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/preview/CassandraMessagePreviewModule.java

```diff
@@ -0,0 +1,40 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.cassandra.preview;
+
+import static com.datastax.driver.core.DataType.text;
+
+import org.apache.james.backends.cassandra.components.CassandraModule;
+import org.apache.james.backends.cassandra.utils.CassandraConstants;
+import org.apache.james.jmap.cassandra.preview.table.CassandraMessagePreviewTable;
+
+import com.datastax.driver.core.schemabuilder.SchemaBuilder;
+
+public interface CassandraMessagePreviewModule {
+    CassandraModule MODULE = CassandraModule.table(CassandraMessagePreviewTable.TABLE_NAME)
+        .comment("Storing the the JMAP preview property extracted from message bodies")
+        .options(options -> options
+            .caching(SchemaBuilder.KeyCaching.ALL,
+                SchemaBuilder.rows(CassandraConstants.DEFAULT_CACHED_ROW_PER_PARTITION)))
+        .statement(statement -> statement
+            .addPartitionKey(CassandraMessagePreviewTable.MESSAGE_ID, text())
```

Comment:
message id is probably an UUID

Reply:
Zell that depends of the mailbox implementation, which don't necessarilly relies on UUID.

From here we can only expect a serializable MessageId. Sad...

Reply:
Do you mean we can have a CassandraMessagePreview with a impl of Mailbox that doesn't use Cassandra?

Reply:
Well I'm reluctant to link server/data-jmap-cassandra to mailbox/cassandra, that don't look much like an orthogonal architecture friendly decision...

Reply:
BTW, there is another test issue where the contract uses another Id (TestId) and it cannot be casted to CassandraId in the DAO (which is hard to fix)

Reply:
> BTW, there is another test issue where the contract uses another Id (TestId) and it cannot be casted to CassandraId in the DAO (which is hard to fix)

Unless the contract specifies a `MessageId generateNewMessageId()` methods that implementation should override

Reply:
> Well I'm reluctant to link server/data-jmap-cassandra to mailbox/cassandra, that don't look much like an orthogonal architecture friendly decision...

I would agree if MessageId were not a problem. If it's an incremental integer, you'll run into troubles regarding Cassandra distribution of entries.

If you want to make it play well with "any mailbox implementation that produces sensible MessageId" then we should enforce that property, for example by publishing the underlying type of MessageId to guice and requiring it to be UUID-based in data-jmap-cassandra.

It makes things more complex but flexibility comes at a cost.

Reply:
so, no UUID?